python-lib/dku_data_processing/format.py

```diff
@@ -0,0 +1,164 @@
+        df[new_detail_column] = 1
+    else:
+        df[new_detail_column] = normalize(df[detail].values)
+    if (detail is not None) and (df.dtypes[detail] not in ['int64', 'float64']):
+        raise ValueError("The detail column should be a int or float type.")
+
+    def convert(x):
+        if x is not None:
+            return float(x)
+        else:
+            return None
+
+    new_longitude_column = get_new_column_name(df, "longitude")
+    new_latitude_column = get_new_column_name(df, "latitude")
+
+    # Parse longitude and latitude from WKT geopoint column
+    parsed_lat_long_df = df[geopoint].apply(wkt_parser)
+    df[new_longitude_column] = parsed_lat_long_df.apply(lambda x: x[0]).apply(lambda x: convert(x))
+    df[new_latitude_column] = parsed_lat_long_df.apply(lambda x: x[1]).apply(lambda x: convert(x))
+
+    # Handle tooltip
+    tooltip_column_to_keep = set()
+    if tooltips:
+        for col_ in tooltips:
+            tooltip_column_to_keep.add(col_['column'])
+    tooltip_column_to_keep = list(tooltip_column_to_keep)
+    tooltip_df = df[tooltip_column_to_keep]
+
+    df = df[[new_latitude_column, new_longitude_column, new_detail_column]]
+    new_tooltip_column = get_new_column_name(df, 'tooltip')
+    df[new_tooltip_column] = tooltip_df.to_dict(orient='records')
+    df = df.rename(columns={new_latitude_column: "lat", new_longitude_column: "long",
+                            new_detail_column: "detail", new_tooltip_column: "tooltip"})
+    df = df.dropna(subset=['lat', 'long'])
+
+    geodata_object = df.to_dict(orient='records')
+
+    return geodata_object
+
+
+def get_geodata_from_dataset(dss_dataset, geopoint, detail, tooltips, filters):
+    """
+    Get the selected geospatial data that will be sent to front-end based on
+    necessary columns.
+
+    :param dss_dataset: DSS dataset from which values are extracted
+    :param geopoint:
+    :param detail:
+    :param tooltips:
+    :param filters:
+    :return:
+    """
+    schema = dss_dataset.read_schema()
+    column_names = [dss_column['name'] for dss_column in schema]
+    columns_to_retrieve = set()
+
+    if geopoint is None:
+        logger.warning("The column containing the geopoint must be specified.")
+        return []
+
+    if geopoint not in column_names:
+        logger.warning("The column containing the geopoint must be part of the dataset.")
+        return []
+
+    columns_to_retrieve.add(geopoint)
+
+    if (tooltips is not None) and (tooltips != []):
+        for col_ in tooltips:
+            columns_to_retrieve.add(col_['column'])
+
+    if detail is not None:
+        columns_to_retrieve.add(detail)
+
+    if (filters is not None) and (filters != []):
+        for col_ in filters:
+            columns_to_retrieve.add(col_['column'])
+
+    # Sampling should be the same as DSS by default
+    df = dss_dataset.get_dataframe(sampling='head', limit=10000, columns=columns_to_retrieve)
```

Comment:
Fix for error white date filtering
```suggestion
    df = dss_dataset.get_dataframe(sampling='head', limit=10000)
    df = df[list(columns_to_retrieve)]
```
It seems that it is DSS Api that has a problem, this way will always work

Reply:
Done in bdaadc9